python/ml4ir/base/pipeline.py

```diff
@@ -413,13 +394,15 @@ def run(self):
                 )
 
             # Finish
-            self.finish(job_status, job_info)
+            # self.finish(job_status, job_info)
 
         except Exception as e:
             self.logger.error("!!! Error Training Model: !!!\n{}".format(str(e)))
             traceback.print_exc()
             job_status = "_FAILURE"
             job_info = "{}\n{}".format(str(e), traceback.format_exc())
+            self.finish(job_status, job_info)
+            return self
```

Comment:
the code should exit here by calling finish to populate the 'FAILURE' file with the exception. Otherwise, it continues and tries to calculate scores.

Reply:
Yes! Again, bad rebase.